qa/rpc-tests/test_framework/mininode.py

```diff
@@ -359,57 +359,131 @@ def __repr__(self):
             % (repr(self.prevout), bytes_to_hex_str(self.scriptSig),
                self.nSequence)
 
-def setCTxOutValue(amount):
-    vchCommitment = [0]*33
-    for i in range(8): #8 bytes
-        vchCommitment[33-1-i] = ((amount >> (i*8)) & 0xff)
-    return ''.join(map(chr, vchCommitment))
+class CTxOutAsset(object):
```

Comment:
the C++ version of these 3 classes are a single template. Perhaps these should be merged as well and use variables instead of template parameters.